File: modules/angular2/src/web_workers/ui/xhr_impl.ts
import {Injectable} from 'angular2/src/core/di';
import {PRIMITIVE} from 'angular2/src/web_workers/shared/serializer';
import {XHR_CHANNEL} from 'angular2/src/web_workers/shared/messaging_api';
import {XHR} from 'angular2/src/core/compiler/xhr';
import {ServiceMessageBrokerFactory} from 'angular2/src/web_workers/shared/service_message_broker';
import {bind} from './bind';

@Injectable()
export class MessageBasedXHRImpl {
  constructor(private _brokerFactory: ServiceMessageBrokerFactory, private _xhr: XHR) {}

  start(): void {
    var broker = this._brokerFactory.createMessageBroker(XHR_CHANNEL);
    broker.registerMethod("get", [PRIMITIVE], bind(this._xhr.get, this._xhr), PRIMITIVE);
  }
}

File: lib/src/web_workers/ui/xhr_impl.dart
library angular2.src.web_workers.ui.xhr_impl;

import "package:angular2/src/core/di.dart" show Injectable;
import "package:angular2/src/web_workers/shared/serializer.dart" show PRIMITIVE;
import "package:angular2/src/web_workers/shared/messaging_api.dart"
    show XHR_CHANNEL;
import "package:angular2/src/compiler/xhr.dart" show XHR;
import "package:angular2/src/web_workers/shared/service_message_broker.dart"
    show ServiceMessageBrokerFactory;
import "bind.dart" show bind;

@Injectable()
class MessageBasedXHRImpl {
  ServiceMessageBrokerFactory _brokerFactory;
  XHR _xhr;
  MessageBasedXHRImpl(this._brokerFactory, this._xhr) {}
  void start() {
    var broker = this._brokerFactory.createMessageBroker(XHR_CHANNEL);
    broker.registerMethod(
        "get", [PRIMITIVE], bind(this._xhr.get, this._xhr), PRIMITIVE);
  }
}
